src/Bonsai.ML/PythonHelper.cs

```diff
@@ -0,0 +1,120 @@
+using Python.Runtime;
+using System;
+using System.Collections.Generic;
+
+namespace Bonsai.ML
+{
+    public class PythonHelper
+    {
+        public static object GetPythonAttribute(PyObject pyObject, string attributeName)
+        {
+            if (pyObject.HasAttr(attributeName))
+            {
+                using (var attr = pyObject.GetAttr(attributeName))
+                {
+                    return ConvertPythonObjectToCSharp(attr);
+                }
+            }
+            else
+            {
+                throw new InvalidOperationException($"Attribute {attributeName} not found in Python object.");
+            }
+        }
+
+        public static T GetPythonAttribute<T>(PyObject pyObject, string attributeName)
+        {
+            if (pyObject.HasAttr(attributeName))
+            {
+                using (var attr = pyObject.GetAttr(attributeName))
+                {
+                    return (T)attr.AsManagedObject(typeof(T));
+                }
+            }
+            else
+            {
+                throw new InvalidOperationException($"Attribute {attributeName} not found in Python object.");
+            }
+        }
+
+        public static object ConvertPythonObjectToCSharp(PyObject pyObject)
```

Comment:
I would be curious to discuss how this is being used in more detail, since there might be opportunities here to keep type-checking, rather than making the output type be the `object` class.